Sources/ParseLiveQuery/Client.swift

```diff
@@ -149,13 +150,13 @@ extension Client {
             )
             subscriptions.append(subscriptionRecord)
 
-            if socket == nil {
-                if !disconnected {
-                    reconnect()
-                }
-            } else {
-                sendOperationAsync(.Subscribe(requestId: subscriptionRecord.requestId, query: query))
-            }
+        if socket == nil || socket?.readyState == .CLOSED{
```

Comment:
Also, I believe you don't need to check for `.CLOSED`, but rather to check for `!= .OPEN`, since if you are in the process of closing the socket - it won't reopen it, whilst it should.